src/core/protocol-authorization.ts

```diff
@@ -77,9 +77,18 @@ export class ProtocolAuthorization {
     incomingMessage: RecordsWrite,
     messageStore: MessageStore,
   ): Promise<void> {
-    // fetch ancestor message chain
-    const ancestorMessageChain: RecordsWriteMessage[] =
-      await ProtocolAuthorization.constructAncestorMessageChain(tenant, incomingMessage, incomingMessage, messageStore);
+    const existingInitialWrite = await ProtocolAuthorization.fetchInitialWrite(tenant, incomingMessage.message.recordId, messageStore);
+
+    let descendantRecordId;
```

Comment:
NIT:  I understand why `descendantRecordId ` is named such within the context of `constructRecordChain`, is there a better name for it within this context?

Feels strange calling it a `descendentRecordId` while setting it to the `parentId` of the incoming message.

Reply:
I thought hard and couldn't come up with something better... we could do this also if you prefer:
```suggestion
    let recordChain;
    if (existingInitialWrite === undefined) {
      recordChain = await ProtocolAuthorization.constructRecordChain(tenant, incomingMessage.message.descriptor.parentId, messageStore);
    } else {
      recordChain = await ProtocolAuthorization.constructRecordChain(tenant, incomingMessage.message.recordId, messageStore);
    }
```

Reply:
Yeh I'm fine with the change you proposed, or calling it something like `rootRecordId` and a comment is also fine.